Sources/MongoSwift/SDAM.swift

```diff
@@ -214,7 +238,12 @@ public struct TopologyDescription {
     public let type: TopologyType
 
     /// The replica set name.
-    public var setName: String? { return self.servers[0].setName }
+    public var setName: String? {
+        guard !self.servers.isEmpty else {
```

Comment:
this would crash on empty self.servers 😬

Reply:
:man_facepalming: